app/src/openxr/cpp/DeviceDelegateOpenXR.cpp

```diff
@@ -713,6 +714,35 @@ struct DeviceDelegateOpenXR::State {
     passthroughCreateInfo.flags = XR_PASSTHROUGH_IS_RUNNING_AT_CREATION_BIT_FB;
     CHECK_XRCMD(OpenXRExtensions::sXrCreatePassthroughFB(session, &passthroughCreateInfo, &passthrough));
   }
+
+  void HandlePassthroughEvent(const XrEventDataPassthroughStateChangedFB& event) {
+    passthroughState = event.flags;
+
+    if ((passthroughState & XR_PASSTHROUGH_STATE_CHANGED_REINIT_REQUIRED_BIT_FB) ||
+        (passthroughState & XR_PASSTHROUGH_STATE_CHANGED_NON_RECOVERABLE_ERROR_BIT_FB)) {
+      // Destroy and re-create passthrough instance and layer
+      if (passthroughLayer->xrLayer != XR_NULL_HANDLE)
+        passthroughLayer->Destroy();
+
+      if (passthrough != XR_NULL_HANDLE) {
+        CHECK_XRCMD(OpenXRExtensions::sXrDestroyPassthroughFB (passthrough));
+        passthrough = XR_NULL_HANDLE;
+      }
+    }
+
+    if (passthroughState & XR_PASSTHROUGH_STATE_CHANGED_REINIT_REQUIRED_BIT_FB) {
+      InitPassthrough();
+
+      vrb::RenderContextPtr ctx = context.lock();
+      passthroughLayer->Init(javaContext->env, session, ctx);
+
+      // Clear this bit from the flags to avoid handling it again
+      passthroughState &= ~(XR_PASSTHROUGH_STATE_CHANGED_REINIT_REQUIRED_BIT_FB);
```

Comment:
This looks weird, `passthroughState` should be reset every time this function is called. I expect the runtime to give us the right flags.

Also why do we need to store the state in an attribute at all?

Reply:
> This looks weird, passthroughState should be reset every time this function is called. I expect the runtime to give us the right flags.

Here I have doubts. The problem is that in the extension it doesn't specify whether a new state is emitted without the `reinit` flag immediately after it has been signaled. And I can't test that since I have not way to force the runtime to emit it. So to err on the safe side, I decided to remove the flag just in case to avoid reinitializing the objects continuously if the runtime doesn't emit anything after that. Probably we should ask for clarification to the extension spec. 

> Also why do we need to store the state in an attribute at all?

To simplify implementation of `CanEnablePassthrough()` method below.

Reply:
Ah right I missed that you're using it bellow.

In any case I don't quite like the fact that we reset the flags that come from an event. The thing is that you're treating an event as it was an state, and it is not, it's just a notification. As you said we cannot be sure about the runtime sending us events for every change. That's why I think storing those flags as an "state" is not correct, and might be error prone. 

In this case I think it's enough to store a boolean that signals the presence of an error, like hasPassthroughError (or a much better name). You'd flip it to true with  both NON_RECOVERABLE and RECOVERABLE errors and only flip it back to true when receiving `XR_PASSTHROUGH_STATE_CHANGED_RESTORED_ERROR_BIT_FB`. With that there is no need to store the flags nor to clear the flags.

Reply:
Ok, what you propose is sensible and indeed simpler. It is also less dependent on the FB extension as well, which is a plus. Will change it.